src/map/script.cpp

```diff
@@ -11033,6 +11033,34 @@ BUILDIN_FUNC(getmapusers)
 	script_pushint(st,map_getmapdata(m)->users);
 	return SCRIPT_CMD_SUCCESS;
 }
+
+/*==========================================
+* getmapuserslist("mapname")
+* set '.@mapuserslist' array contained the account id of players in the map
+*------------------------------------------*/
+BUILDIN_FUNC(getmapuserslist)
+{
+	struct block_list *bl;
+	struct map_data *mapdata;
+	int bsize, i, j = 0;
+	const char* m = script_getstr(st, 2);
+
+	if (map_mapname2mapid(m)< 0) {
+		ShowWarning("buildin_getmapuserslist: Unknown map '%s'.\n", m);
+		return SCRIPT_CMD_FAILURE;
+	}
+
+	mapdata = map_getmapdata(map_mapname2mapid(m));
+	bsize = mapdata->bxs * mapdata->bys;
+
+	for (i = 0; i < bsize; i++)
+		for (bl = mapdata->block[i]; bl != NULL; bl = bl->next)
+			if (bl->type == BL_PC)
+				setd_sub(st, NULL, ".@mapuserslist", j++, (void *)__64BPRTSIZE(bl->id), NULL);
```

Comment:
Also, this part is incorrect (just I think that, don't be offend). 
Because `.@` vars, super complex one. Better to use global server var not bound to some NPC or a player. 

Like:
```
mapreg_setreg(reference_uid(add_str("$@mapuserslist"), j++), sd->status.char_id);
```
where j++ is an iterator, also sd (map-session) you will get with mapit_getallusers() iterator. So you will set char_ids to your array with player char_ids which easily possible to convert to names if it will be needed.

Reply:
i made it like that first but after testing
using $@variable mean the index wont be reset(reason down) so using it like that mean we need to clear the array each time we want to use the command

(see the first Commit maybe i did something wrong to case this) but
example:
```cs
prontera,153,187,5	script	getmapuserslist_test	444,{
	getmapuserslist("prontera");
	
	copyarray .@accountID, $@mapuserslist, getarraysize($@mapuserslist);
	
	debugmes "map users count : " + getarraysize(.@accountID);
	for (.@i = 0; .@i < getarraysize(.@accountID); .@i++)
		debugmes "account id : " + .@accountID[.@i] + " | player name : " + rid2name(.@accountID[.@i]);
	end;
}
```
using this script with 2 players at prontera well return the 2 players
warp the one of them to another map

use the using the script again would case one of 2

replacing index[0] with the player left in the map and if he was index[1] would stay at index[1] too 
this would return the player twice

if the player was index[0] and the other player was index[1] would return the players again even if one of them outside the map

i don't know why this happen as i did follow the way used by other commands but the variable did not reset (thinking about it again now i think i need to retest that)

so i used .@variable as it's better as onetime variable as the command maybe used in warpers etc so it maybe used a lot

Reply:
> using $@variable mean the index won't be reset(reason down) so using it like that mean we need to clear the array each time we want to use the command

Yea, I know about it, It's solved via:
```
mapreg_setreg(reference_uid(add_str("$@mapuserslist"), 0), 0);
```
before loop. Try it =)

Reply:
still the same issue , we need to loop it to clear it

but anyway i think what anyone should do is copy the `$@var` to `.@var` so i don't see the problem of making it `.@var` directly